### PR TITLE
Remove Axis from public API

### LINKED ISSUE #405: Lets get rid of `Axis`
I propose that as part of our current substantial refactor, we eliminate the `Axis` object altogether!

What we have to gain:
- 1200+ lines of code we do not have to maintain anymore. 🤯

What we loose: 
- Pretty much nothing! @TomNicholas work will make the axis methods obsolete, and the the two remaining aspects (storing axis attributes + `transform`) can comfortably live in the grid object IMO. 
    - The axis attributes can be stored just as easy as dictionaries `{axisname:value, ...}` in the grid object directly (actually certain things like #404 are easier this way). Similarly in #399 I actually ended up storing the `face_connections` on the grid.
    - I am confident that the transform functionality is not to hard to migrate, since it is pretty separate from everything else.

What do people think? @rabernat did I miss something here?

### PR DESCRIPTION
**Pull request by TomNicholas**
Removing huge amounts of code that the refactor makes redundant! Early stage so far but wanted to push so that it was backed up somewhere other than my laptop.

We decided not to remove the `Axis` class completely, as the abstraction is still useful internally, but we did remove it from public API. That means we were free to remove all the Axis computation methods.

Also we tried changing `Axis` to a `dataclass` but found it was more trouble than it was worth.

 - [x] Will close #405, and #406
 - [ ] Tests added
 - [ ] Passes `pre-commit run --all-files`
 - [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
 - [ ] New functions/methods are listed in `api.rst`

### CODE REVIEW

File: xgcm/grid.py [hunk 1]
```diff
@@ -37,106 +40,42 @@ def _maybe_promote_str_to_list(a):
...
-
-         Outer
-         |------o-------|------o-------|------o-------|------o-------|
-        [0]            [1]            [2]            [3]            [4]
-
-    The `center` position is the only one without the `c_grid_axis_shift`
-    attribute, which must be present for the other four. However, the actual
-    value of `c_grid_axis_shift` is ignored for `inner` and `outer`, which are
-    differentiated by their length.
-    """
```
Comment (TomNicholas) [inline xgcm/grid.py:70]: Probably want to find a better place to preserve the nice explanation in this docstring.

**Comment by jbusecke** [comment]
Oh yeahhhh. I am excited for this. Ill take a deeper look soon.

**Comment by jbusecke** [comment]
Will this be superseeded by #557 entirely? @TomNicholas

**Comment by jbusecke** [comment]
Closing as superseeded by #557

**Pull request closed without merging**